src/messageServer.js

```
@@ -0,0 +1,32 @@
+import { messageService } from './services/message.service.js';
+
+export const messageServer = (io) => {
+  io.on('connection', (socket) => {
+    // eslint-disable-next-line no-console
+    console.log('Новий користувач підключився:', socket.id);
+
+    socket.on('joinRoom', ({ roomId }) => {
+      socket.join(`room_${roomId}`);
+    });
+
+    socket.on('newMessage', async ({ roomId, userId, text }) => {
+      try {
+        const message = await messageService.createMessage({
+          roomId,
+          userId,
+          text,
+        });
+
+        io.to(`room_${roomId}`).emit('messageBroadcast', message);
+      } catch (err) {
+        // eslint-disable-next-line no-console
+        console.error('Error in newMessage handler:', err);
```

Error handling inside the `newMessage` handler only logs to console. Consider informing the sender about failure (ack or an error event), e.g., `socket.emit('messageError', { error: '...' })` or use callback acknowledgements in Socket.IO. This helps clients surface errors to users instead of failing silently on the server.